apps/volk_profile.cc

```diff
@@ -142,6 +151,42 @@ int main(int argc, char* argv[])
     }
 
 
+    // Print session summary
+    if (results.size() > 0) {
+        double sum_speedup = 0.0;
+        double max_speedup = 0.0;
+        std::string max_speedup_kernel;
+        int kernel_count = 0;
+
+        for (const auto& r : results) {
+            auto generic_it = r.results.find("generic");
+            if (generic_it == r.results.end())
+                continue;
+            double generic_time = generic_it->second.time;
+
+            // Use best aligned arch for speedup
+            auto best_it = r.results.find(r.best_arch_a);
+            if (best_it != r.results.end() && r.best_arch_a != "generic") {
+                double speedup = generic_time / best_it->second.time;
+                sum_speedup += speedup;
+                if (speedup > max_speedup) {
+                    max_speedup = speedup;
+                    max_speedup_kernel = r.name;
+                }
+                kernel_count++;
+            }
+        }
+
+        if (kernel_count > 0) {
+            std::cout << std::endl;
+            std::cout << "Session summary (" << kernel_count << " kernels):" << std::endl;
+            std::cout << "  Average speedup: " << std::fixed << std::setprecision(2)
+                      << (sum_speedup / kernel_count) << "x" << std::endl;
+            std::cout << "  Max speedup: " << max_speedup << "x (" << max_speedup_kernel
+                      << ")" << std::endl;
```

Comment:
I like this summary. Though, when I read through your PR explanation, I wondered what it compares to? A previous run? Some other kernel?
`generic` makes sense. Can you add that in the description?